src/environmentd/src/http.rs

```diff
@@ -123,38 +123,53 @@ impl HttpServer {
-            metrics,
-        })
-        .layer(DefaultBodyLimit::max(MAX_REQUEST_SIZE))
-        .layer(middleware::from_fn(move |req, next| {
-            let base_frontegg = Arc::clone(&base_frontegg);
-            async move { http_auth(req, next, tls_mode, &base_frontegg).await }
-        }))
-        .layer(Extension(adapter_client_rx.shared()))
-        .layer(Extension(Arc::clone(&active_connection_count)))
-        .layer(
-            CorsLayer::new()
-                .allow_credentials(false)
-                .allow_headers([
-                    AUTHORIZATION,
-                    CONTENT_TYPE,
-                    HeaderName::from_static("x-materialize-version"),
-                ])
-                .allow_methods(Any)
-                .allow_origin(allowed_origin)
-                .expose_headers(Any)
-                .max_age(Duration::from_secs(60) * 60),
-        );
+        let base_router = base_router(BaseRouterConfig { profiling: false })
+            .layer(middleware::from_fn(move |req, next| {
+                let base_frontegg = Arc::clone(&base_frontegg);
+                async move { http_auth(req, next, tls_mode, &base_frontegg).await }
+            }))
+            .layer(Extension(adapter_client_rx.shared()))
+            .layer(Extension(Arc::clone(&active_connection_count)))
+            .layer(
+                CorsLayer::new()
+                    .allow_credentials(false)
+                    .allow_headers([
+                        AUTHORIZATION,
+                        CONTENT_TYPE,
+                        HeaderName::from_static("x-materialize-version"),
+                    ])
+                    .allow_methods(Any)
+                    .allow_origin(allowed_origin)
+                    .expose_headers(Any)
+                    .max_age(Duration::from_secs(60) * 60),
+            );
+
         let ws_router = Router::new()
             .route("/api/experimental/sql", routing::get(sql::handle_sql_ws))
             .with_state(WsState {
                 frontegg,
-                adapter_client,
+                adapter_client: adapter_client.clone(),
                 active_connection_count,
             });
-        let router = Router::new().merge(base_router).merge(ws_router);
+
+        let webhhook_router = Router::new()
+            .route(
+                "/api/webhook/:database/:schema/:id",
+                routing::post(webhook::handle_webhook),
+            )
+            .with_state(adapter_client);
+
+        let router = Router::new()
+            .merge(base_router)
+            .merge(ws_router)
+            .merge(webhhook_router)
+            // Layers that get applied for __all__ routes.
+            //
+            // Note: if there are layers that should only get applied for certain routes, add them
+            // to specific routers.
+            .layer(DefaultBodyLimit::max(MAX_REQUEST_SIZE))
+            .layer(metrics::PrometheusLayer::new(metrics));
```

Comment:
Since these layers are applied to both the internal and external HTTP servers, can we factor out into a function like `apply_default_layers` that's called in both code paths? Hopefully the types in that function's signature are not too terrible.

Reply:
Good idea! I created a trait called `DefaultLayers` with one method `apply_default_layers`, this way we could continue to use the builder pattern of chaining calls